src/D_functions_and_procedures.sql

```diff
@@ -146,3 +146,40 @@ BEGIN
     PERFORM rate(john_id, 'tt1375666', 9.0); 
     PERFORM rate(jane_id, 'tt1375666', 7.0); 
 END $$;
+
+-- D7
+CREATE OR REPLACE FUNCTION calculate_name_rating() RETURNS VOID AS $$
+DECLARE
+    name_id INT;
+    weighted_rating DECIMAL(5, 2);
+BEGIN
+    -- Iterate through all persons (actors and crew members)
+    FOR name_id IN (SELECT person_id FROM person) LOOP
+        -- Calculate the weighted average rating for each person (both actors and crew members)
+        SELECT SUM(sc.value * sc.vote_count) / SUM(sc.vote_count)
+        INTO weighted_rating
+        FROM score sc
+        INNER JOIN cast_member cm ON sc.media_id = cm.media_id
+        WHERE cm.person_id = name_id
+        
+        UNION ALL
```

Comment:
I have a question regarding the use of UNION ALL and the use of INTO more than ones. Because I run the function for fun but I get an error saying that: 

*INTO specified more than once at or near "INTO"* so my question is, is this okay?

Reply:
You were right! I never actually ran it 😃